packages/main/src/ComboBox.ts

```diff
@@ -384,16 +148,263 @@ const metadata = {
+	 * </ul>
+	 *
+	 * @type {sap.ui.webc.base.types.ValueState}
+	 * @name sap.ui.webc.main.ComboBox.prototype.valueState
+	 * @defaultvalue "None"
+	 * @public
+	 */
+	@property({ type: ValueState, defaultValue: ValueState.None })
+	valueState!: ValueState;
+
+	/**
+	 * Defines whether the component is read-only.
+	 * <br><br>
+	 * <b>Note:</b> A read-only component is not editable,
+	 * but still provides visual feedback upon user interaction.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.ComboBox.prototype.readonly
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	readonly!: boolean;
+
+	/**
+	 * Defines whether the component is required.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.ComboBox.prototype.required
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	required!: boolean;
+
+	/**
+	 * Indicates whether a loading indicator should be shown in the picker.
+	 *
+	 * @type {boolean}
+	 * @name sap.ui.webc.main.ComboBox.prototype.loading
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	loading!: boolean;
+
+	/**
+	 * Defines the filter type of the component.
+	 * Available options are: <code>StartsWithPerTerm</code>, <code>StartsWith</code> and <code>Contains</code>.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.ComboBox.prototype.filter
+	 * @defaultvalue "StartsWithPerTerm"
+	 * @public
+	 */
+	@property({ type: ComboBoxFilter, defaultValue: ComboBoxFilter.StartsWithPerTerm })
+	filter!: ComboBoxFilter;
+
+	/**
+	 * Indicates whether the input is focssed
+	 * @private
+	 */
+	 @property({ type: Boolean })
+	 focused!: boolean;
+
+	/**
+	 * Indicates whether the visual focus is on the value state header
+	 * @private
+	 */
+	@property({ type: Boolean })
+	_isValueStateFocused!: boolean;
+
+	/**
+	 * Defines the accessible ARIA name of the component.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.ComboBox.prototype.accessibleName
+	 * @defaultvalue: ""
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property()
+	accessibleName!: string;
+
+	/**
+	 * Receives id(or many ids) of the elements that label the component
+	 * @type {string}
+	 * @name sap.ui.webc.main.ComboBox.prototype.accessibleNameRef
+	 * @defaultvalue ""
+	 * @public
+	 * @since 1.0.0-rc.15
+	 */
+	@property({ defaultValue: "" })
+	accessibleNameRef!: string;
+
+	@property({ type: Boolean, noAttribute: true })
+	_iconPressed!: boolean;
+
+	@property({ type: Object, noAttribute: true, multiple: true })
+	_filteredItems!: Array<ComboBoxItem | ComboBoxGroupItem>;
+
+	@property({ validator: Integer, noAttribute: true })
+	_listWidth?: number | undefined;
+
+	/**
+	 * Defines the component items.
+	 *
+	 * @type {sap.ui.webc.main.IComboBoxItem[]}
+	 * @name sap.ui.webc.main.ComboBox.prototype.default
+	 * @slot items
+	 * @public
+	 */
+	@slot({ type: HTMLElement, "default": true, invalidateOnChildChange: true })
+	items!: Array<ComboBoxItem | ComboBoxGroupItem>;
```

Comment:
Better we introduce an interface IComboBoxItem that both ComboBoxItem & ComboBoxGroupItem implement. There are examples in other components.